lib/blocks.php

```diff
@@ -296,3 +296,197 @@ function gutenberg_enqueue_editor_block_styles_assets() {
 	}
 	add_action( 'enqueue_block_editor_assets', 'gutenberg_enqueue_editor_block_styles_assets' );
 }
+
+/**
+ * Renders the classNames and styles for blocks
+ *
+ * @param string $block_content Output of the current block.
+ * @param array $block Block Object.
+ * @return string New block output.
+ */
+function gutenberg_experimental_apply_classnames_and_styles( $block_content, $block ) {
+	// Don't filter template part blocks since we filter the blocks in each template
+	// part individually.
+	if ( 'core/template-part' == $block[ 'blockName' ] ) {
+		return $block_content;
+	}
+
+	// TODO: Check for supports: _experimental* before adding any of this to a block.
```

Comment:
seems like the checks could be added to the underlying functions (`gutenberg_experimental_build_css_colors`...)

Reply:
Added some of the checks in.  Currently were getting supports array in the main function and passing it to the others to make the checks as necessary.